synapticsmst: Modify method to access board ID

superm1 opened the pull request:
In some firmware images HDCP 2.2 will be enabled by default which means that access to the flash region will fail.

This PR sorts that out.

Note: this is actually to 1_1_X since that's what @ryanchang0624 did the work on, but it can easily cherry pick to master after merging.

superm1 commented on plugins/synapticsmst/synapticsmst-device.c:
```diff
@@ -300,13 +301,19 @@ synapticsmst_device_read_board_id (SynapticsMSTDevice *device,
 		}
+		byte[0] = (dwData[0]>>8)&0x000000ff;
+		byte[1] = dwData[0]&0x000000ff;		
+		g_debug ("0x170E: %lx bID %x cID %x", dwData[0], byte[0], byte[1]);
```
Since now outputting this information in the `synapticsmst_read_board_id` function I think that the other debug line (381 in this PR) can probably be removed now.

ryanchang0624 replied:
Fixed in latest push.

superm1 commented on plugins/synapticsmst/synapticsmst-device.c:
```diff
@@ -264,6 +264,7 @@ synapticsmst_device_read_board_id (SynapticsMSTDevice *device,
 				   guint8 *byte,
 				   GError **error)
 {
+	guint32 dwData[4] = {0};
```
Shouldn't this be guint8 dwData[4] = {0} ?  You only need 4 bytes it seems.

ryanchang0624 replied:
Fixed in latest push.

superm1 commented:
@ryanchang0624:
Can you please adjust the commit message for this?  The title should include the prefix of the component changed and the description should explain why for the change.  You can use `git commit --amend` on your commit on your branch to fix this.

Example title: 

> synapticsmst: Access Board ID through memory instead

Example description:

> In some firmware images HDCP 2.2 will be enabled by default which means that access to the flash region will fail.

superm1 commented on plugins/synapticsmst/synapticsmst-device.c:
```diff
@@ -377,7 +384,7 @@ synapticsmst_device_enumerate_device (SynapticsMSTDevice *device,
 	if (!synapticsmst_device_read_board_id (device, connection, byte, error))
 		return FALSE;
-	priv->board_id = (byte[0] << 8) | (byte[1]);
+	priv->board_id = (byte[1] << 8) | (byte[0]);
```
I think this is likely going to cause CI to fail since the test suite (with captured files) inverted the order of these bytes.  So might need to flip the bytes filled in line 313 and 314 of this PR.

ryanchang0624 replied:
Fixed in latest push.

superm1 commented:
I guess `ADDR_CUSTOMER_ID` and `ADDR_BOARD_ID` can now be removed too right?

superm1 commented:
Looks like CI is failing too.
```
../plugins/synapticsmst/synapticsmst-device.c: In function ‘synapticsmst_device_read_board_id’:
../plugins/synapticsmst/synapticsmst-device.c:315:12: error: format ‘%lx’ expects argument of type ‘long unsigned int’, but argument 4 has type ‘guint32’ {aka ‘unsigned int’} [-Werror=format=]
   g_debug ("0x170E: %lx bID %x cID %x", dwData[0], byte[0], byte[1]);
```

ryanchang0624 commented:
> I guess `ADDR_CUSTOMER_ID` and `ADDR_BOARD_ID` can now be removed too right?

Still needed to get the ID in payload to compare with ID in memory when write firmware.

ryanchang0624 commented:
@superm1 I have modified the function and push it.

superm1 commented on plugins/synapticsmst/synapticsmst.quirk:
```diff
@@ -15,7 +15,7 @@
 #
 
-[SynapticsMSTBoardID=0]
+[SynapticsMSTBoardID=2]
```
Is this intended change?

superm1 replied:
I think this should be only pending question.  Is this intended change? (Eg should EVB be board ID 2 for new board address reading procedure)?

If so, there are a few other places in the code that treat EVB specially and should be adjusted too.

superm1 commented on plugins/synapticsmst/synapticsmst-device.c:
```diff
@@ -300,11 +300,13 @@ synapticsmst_device_read_board_id (SynapticsMSTDevice *device,
-							 error)) {
-			g_prefix_error (error, "failed ot read EEPROM: ");
+							UPDC_READ_FROM_MEMORY,
+							4,
```
Since relevant data is in the first two bytes can you just read those?

superm1 commented:
It looks like CI is broken due to some unused variables from last change.  Can you sort those out (`code_size`, `offset`, `checksum`, `payload_data`)?

ryanchang0624 commented:
Fixed already, apologize for the mistake.

From: Mario Limonciello <notifications@github.com>
Sent: Friday, November 2, 2018 9:31 PM
To: hughsie/fwupd <fwupd@noreply.github.com>
Cc: Ryan Chang <Ryan.Chang@synaptics.com>; Mention <mention@noreply.github.com>
Subject: Re: [hughsie/fwupd] synapticsmst: Modify method to access board ID (#821)


It looks like CI is broken due to some unused variables from last change. Can you sort those out (code_size, offset, checksum, payload_data)?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/hughsie/fwupd/pull/821#issuecomment-435380712>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AmZR2o6N1FZHouudkGjt2B4YqLkXuSTNks5urEj-gaJpZM4X_EGM>.

superm1 commented:
Thanks! I've squashed your commits down in a separate PR for you (I couldn't write to your branch).
https://github.com/hughsie/fwupd/pull/832

I also adjusted the handling for EVB, hope it's OK the change I made.
I'll let @hughsie give a final thumbs up on that branch after CI runs.
